src/main/javascript/src/store/participants.module.js

```diff
@@ -0,0 +1,77 @@
+import { participantService } from "@/services";
+
+const state = {
+  participants: null,
+  groups: null,
+};
+
+const actions = {
+  fetchParticipants: ({ commit }, experimentId) => {
+    return participantService
+      .getAll(experimentId)
+      .then((data) => {
+        commit("setParticipants", data);
+      })
+      .catch((response) => {
+        console.log("fetchParticipants | catch", { response });
+      });
+  },
+
+  updateParticipantsGroup: ({ commit }, participantsList) => {
+    commit("updateParticipantsGroup", participantsList);
+  },
+
+  updateParticipants: ({ state }, experimentId) => {
+    const requestBody = [];
+    state.participants.map((participant) => {
+      const participantDetail = {
+        participantId: participant.participantId,
+        consent: participant.consent,
+        dropped: participant.dropped,
+        groupId: participant.groupId,
+      };
+      requestBody.push(participantDetail);
+    });
+
+    return participantService
+      .updateParticipants(experimentId, requestBody)
+      .catch((response) =>
+        console.log("updateParticipants | catch", { response })
+      );
+  },
+
+  fetchGroups: ({ commit }, experimentId) => {
+    return participantService
+      .getGroups(experimentId)
+      .then((data) => {
+        commit("setGroups", data);
+      })
+      .catch((response) => {
+        console.log("fetchParticipants | catch", { response });
+      });
+  },
+};
+
+const mutations = {
+  setParticipants(state, data) {
+    state.participants = data;
+  },
+
+  updateParticipantsGroup(state, data) {
```

Comment:
Since this is the same type of functionality as `setParticipants` - we should consider a consistent nomenclature. Consider renaming `updateParticipantsGroup` to `setParticipantsGroup`.